Sources/SwifterSwift/SwiftStdlib/CaseIterableExtensions.swift

```diff
@@ -0,0 +1,45 @@
+// CaseIterableExtensions.swift - Copyright 2023 SwifterSwift
+
+import Foundation
+
+public extension CaseIterable where Self: Equatable, AllCases == [Self] {
+    
+    /// SwifterSwift: Sets the value to next available case from `allCases` array.
+    ///
+    ///     enum ShuffleMode: CaseIterable {
+    ///        case off, songs, albums
+    ///     }
+    ///
+    ///     var mode: ShuffleMode = .off
+    ///     mode.toggle() // .songs
+    ///
+    ///     var mode: ShuffleMode = .albums
+    ///     mode.toggle() // .off
+    ///
+    ///     var mode: ShuffleMode = .albums
+    ///     mode.toggle(cycle: false) // .albums
+    ///
+    ///     var mode: ShuffleMode = .albums
+    ///     mode.toggle(between: [.songs, .albums]) // .songs
+    ///
+    /// - Parameters:
+    ///    - cycle: Indicates whether the value should be modified if it's already at the end of `values` array.
+    ///
+    mutating func toggle(
```

Comment:
I understand the reason for this but that seems too specific for extensions on a generic type like `CaseIterable`. This to me seems the kind of problem we solve with a different abstraction e.g. `CircularList`
```swift
let cl = CircularList(sequence: ShuffleMode.allCases)
cl.advanceTo(element: .off)
cl.next() // .songs
```
The meaning of exactly what this code is doing is very clear just reading it whether:
```swift
var mode: ShuffleMode = .off
mode.toggle() // Reading this code, what does toggle mean? 
```
It is just not clear at point of use. So IMO I would go into a different abstraction for this use case.

Reply:
I think that the issue is mostly in the naming, as "toggling" a value is for switching a value on or off, like you have with `Bool`, or you could use it with `UISwitch`, if you wanted to add an extension there.
If you simply renamed this to `nextCase` then I think it could make sense for a `CaseIterable` value to have this function.

Reply:
You can be more generic with this solution (no need for `Array`)
```swift
public extension CaseIterable where Self: Equatable {
    func nextCase(cycle: Bool = false) -> Self {
        let values = Self.allCases
        guard let currentIndex = values.firstIndex(of: self) else {
            fatalError("Case \(self) must be included in `values` array.")
        }
        var index = values.index(after: currentIndex)
        if index == values.endIndex {
            if cycle {
                index = values.startIndex
            } else {
                index = currentIndex
            }
        }
        return values[index]
    }
}
```

Reply:
Sound fair enough, 
Only one other point I would add is for cycle `false` is a bit odd for calling next on last element and for that operation to be no-op without caller receiving a result. Meaning calling that api would advance or not but there is no way for me as a caller to know that other than kind of do compare after. 
Maybe a case to have two functions one for cycle that returns non-optional because it would always have a value and the non-cycle returns nil on next for the last element?

Reply:
Good point about the no-op. So maybe we should have `nextCase` returning `Self?` and `nextCaseCycled` returning `Self`. WDYT @wiencheck?